node/bft-consensus/src/lib.rs

```diff
@@ -169,8 +174,104 @@ impl ExecutionState for MyExecutionState {
                 consensus_output.batches.len(),
                 consensus_output.sub_dag.leader.header.author,
             );
-            // TODO: get the output to snarkOS
-            // self.node.save_consensus(consensus_output).await;
+
+            /*
+            TODO: the following generally works, but there are some open points
+                1. currently, only blocks produced by the beacon are considered valid
+                2. can the Aleo mempool have a different set of transactions than ones accepted by the consensus?
+                3. we can't fail here, i.e. the checks by the TransactionValidator must be final
+                4. every validator can create a block, but should they? the downstream can create it on its own too
+
+            let consensus = self.consensus.clone();
+            let account = self.account.clone();
+            let next_block = tokio::task::spawn_blocking(move || {
+                // Collect all the transactions contained in the agreed upon batches.
+                let mut transactions = Vec::new();
+                for batch in consensus_output.batches {
+                    for batch in batch.1 {
+                        for transaction in batch.transactions {
+                            let bytes = BytesMut::from(&transaction[..]);
+                            // TransactionValidator ensures that the Message can be deserialized.
+                            let message = Message::<N>::deserialize(bytes).unwrap();
```

Comment:
Note: we'll likely need to handle this so that malformed data doesn't break consensus.

Reply:
note the comment :detective:

Reply:
extra background: for a tx to enter this point, it must have been processed by one of the `TransactionValidator` methods, both of which perform deserialization

Reply:
Ah, glazed over the comment for some reason 😄